### ISSUE TITLE
Wrong expected result for BGREWRITEAOF

### ISSUE THREAD

**Issue by israellot** [StackExchange/StackExchange.Redis#1236]
From aof.c on redis
```
void bgrewriteaofCommand(client *c) {
    if (server.aof_child_pid != -1) {
        addReplyError(c,"Background append only file rewriting already in progress");
    } else if (server.rdb_child_pid != -1) {
        server.aof_rewrite_scheduled = 1;
        addReplyStatus(c,"Background append only file rewriting scheduled");
    } else if (rewriteAppendOnlyFileBackground() == C_OK) {
        addReplyStatus(c,"Background append only file rewriting started");
    } else {
        addReply(c,shared.err);
    }
}
```

The StackExchange.Redis client expects a OK response instead.
```
 private ResultProcessor<bool> GetSaveResultProcessor(SaveType type)
        {
            switch (type)
            {
                case SaveType.BackgroundRewriteAppendOnlyFile: return ResultProcessor.DemandOK;
                case SaveType.BackgroundSave: return ResultProcessor.BackgroundSaveStarted;
#pragma warning disable 0618
                case SaveType.ForegroundSave: return ResultProcessor.DemandOK;
#pragma warning restore 0618
                default: throw new ArgumentOutOfRangeException(nameof(type));
            }
        }
```

**Comment by mgravell** [comment]
@antirez - is this an API change?

![image](https://user-images.githubusercontent.com/17328/65624014-ed550e00-dfc0-11e9-80f4-172a235acc31.png)

**Comment by antirez** [comment]
@mgravell it never changed, the documentation was, I believe, *wrong since the start*... Fixing, thank you.

**Comment by antirez** [comment]
@mgravell I just improved the documentation. The description already contained the information that an error could be returned, but now is more explicit, and more cases are documented.

**Comment by NickCraver** [comment]
Tagging this as a bug, we do need a library change to handle all the result types. @antirez thanks for improving the docs!

**Comment by NickCraver** [comment]
This needs some simple string response processor love - raising to take a peek at.

**Comment by NickCraver** [comment]
Fixed PRed in #1978!
